react-backend/client/src/cb-admin/pages/VisitorDetails.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import moment from 'moment';
+import { assocPath, filter, invertObj, map, project, sortBy } from 'ramda';
+import { FlexContainerCol, FlexContainerRow } from '../../shared/components/layout/base';
+import { Heading, Link } from '../../shared/components/text/base';
+import { Form as Fm } from '../../shared/components/form/base';
+import LabelledSelect from '../../shared/components/form/LabelledSelect';
+import TranslucentTable from '../components/TranslucentTable';
+import { Visitors } from '../../api';
+
+
+const Nav = styled.nav`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  flex: 1;
+`;
+
+const FlexItem = styled.div`
+  flex: ${props => props.flex || '1'};
+  height: 100%;
+`;
+
+const Form = Fm.extend`
+  width: 100%;
+`;
+
+const Row = FlexContainerRow.extend`
+  align-content: center;
+  align-items: flex-start;
+  flex: 3;
+`;
+
+const HyperLink = Link.extend`
+  flex: ${props => props.flex || '1'};
+`;
+
+const FormSection = styled.section`
+  width: 20%;
+`;
+
+
+const keyMap = {
+  visit_id: null,
+  visitor_id: 'Visitor ID',
+  gender: 'Gender',
+  yob: 'Year of birth',
+  activity: 'Activities',
+  visit_date: 'Date of visit',
+};
+
+const colToState = invertObj(keyMap);
+
+const columns = Object.values(keyMap).filter(Boolean);
+
+const sortOptions = [{ key: '0', value: '' }].concat(columns.map((col, i) => ({ key: `${i + 1}`, value: col })));
```

Comment:
What's going on here?

Reply:
`keyMap` describes the mapping between keys in state and column display names (hence `visit_id` being `null`).

`colToState` is the reverse mapping.

`columns` is a list of the column display names.

`sortOptions` is the list of things we can sort the table by (i.e. the columns). But in order to be usable by `LabelledSelect` they need to be in a particular form. Also we want a default empty option at the top.